### PR TITLE
Ltcvar

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
Tests require the following https://github.com/OP-TEE/optee_test/pull/11

File: lib/libmpa/include/mpalib.h [hunk 1]
```diff
@@ -105,15 +105,31 @@ typedef mpa_fmm_context_base *mpa_fmm_context;
  * bit_size is the number of bits of the regular vars
  * that this pool was created for.
  */
+/*
```
Comment (jenswi-linaro) [inline lib/libmpa/include/mpalib.h]: Please remove this commented out code.

File: lib/libmpa/mpa_mem_static.c [hunk 2]
```diff
@@ -66,29 +45,54 @@ void mpa_init_scratch_mem(mpa_scratch_mem pool, int nr_vars, int max_bits)
...
-	}
-	if ((4 < tvar->alloc) != 0) {
-		DPRINT("Out of temp vars. Dumping pattern : 0x%X\n",
-		       __mpa_get_alloced_pattern(pool));
-		DPRINT("TOO SMALL SCRATCH MEM AREA. THIS MUST NOT HAPPEN!\n");
-		return NULL;
+	uint32_t offset;
+	size_t size;
+	struct mpa_scratch_item *new_item;
+	struct mpa_scratch_item *last_item = 0;
```
Comment (jenswi-linaro) [inline lib/libmpa/mpa_mem_static.c]: Use NULL instead 0 for pointer assignment.

File: lib/libmpa/mpa_mem_static.c [hunk 3]
```diff
@@ -66,29 +45,54 @@ void mpa_init_scratch_mem(mpa_scratch_mem pool, int nr_vars, int max_bits)
  *  mpa_alloc_static_temp_var
  *
  */
-mpanum mpa_alloc_static_temp_var(mpanum *var, mpa_scratch_mem pool)
+mpanum mpa_alloc_static_temp_var_size(int size_bits, mpanum *var,
```
Comment (jenswi-linaro) [inline lib/libmpa/mpa_mem_static.c]: Please add a description of the memory allocation algorithm, assumptions, limitations and strengths etc.

**Comment by pascal-brand-st-dev** [comment]
Updates with respect to Jens' comments.

**Comment by jenswi-linaro** [comment]
Interesting implementation. It will be easy to extend in the future to work together with the pager in the same way as the thread stacks are handled by the pager today.
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Pull request merged**